pkg/osbuild/machine_id_stage.go

```diff
@@ -0,0 +1,22 @@
+package osbuild
+
+type MachineIdStageOptions struct {
+	// Determines the state of `/etc/machine-id`, valid values are
+	// `yes` (reset to `uninitialized`), `no` (empty), `preserve` (keep).
+	FirstBoot string `json:"first-boot"`
+}
```

Comment:
A set of constants for the stage would be useful here, but not necessary.

Reply:
Fwiw, I agree, having a type and const (like e.g. in qemu_stage.go and the QEMUFormat here) would be slightly nicer as it would give us (more) type safety.